pkg/kstatus/polling/polling.go

```diff
@@ -38,10 +39,20 @@ type StatusPoller struct {
 // back on the event channel returned. The statusPollerRunner can be cancelled at any time by cancelling the
 // context passed in.
 func (s *StatusPoller) Poll(ctx context.Context, identifiers []object.ObjMetadata, options Options) <-chan event.Event {
+	statusReaderFactory := createStatusReaders
```

Comment:
I wish there was a simpler way do to this, but there doesn't seem to be an obvious way do to it. It is a bit unfortunate that that the `StatusReadersFactoryFunc` is a function instead of an interface.

Reply:
I agree it's a little involved but I wanted to introduce this change as lightweight as possible. An interface would be great as a next step, though.